features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/DocumentEnricher.java

```diff
@@ -155,28 +160,42 @@ private static boolean isPrivateAddress(String ipAddress) {
         }
     }
 
-    private Optional<NodeDocument> getNodeInfo(String location, String ipAddress) {
-        return getNodeInfo(location, InetAddressUtils.addr(ipAddress));
+    private Optional<NodeDocument> getNodeInfo(final String location, final String ipAddress, final ContextKey contextKey, final String value) {
+        return getNodeInfo(location, InetAddressUtils.addr(ipAddress), contextKey, value);
     }
 
-    private Optional<NodeDocument> getNodeInfo(String location, InetAddress ipAddress) {
-        final Optional<Integer> nodeId = interfaceToNodeCache.getFirstNodeId(location, ipAddress);
-        if (nodeId.isPresent()) {
-            try (Timer.Context ctx = nodeLoadTimer.time()) {
-                final OnmsNode onmsNode = nodeDao.get(nodeId.get());
-                if (onmsNode != null) {
-                    final NodeDocument nodeInfo = new NodeDocument();
-                    nodeInfo.setForeignSource(onmsNode.getForeignSource());
-                    nodeInfo.setForeignId(onmsNode.getForeignId());
-                    nodeInfo.setNodeId(nodeId.get());
-                    nodeInfo.setCategories(onmsNode.getCategories().stream().map(OnmsCategory::getName).collect(Collectors.toList()));
-
-                    return Optional.of(nodeInfo);
-                } else {
-                    LOG.warn("Node with id: {} at location: {} with IP address: {} is in the interface to node cache, but wasn't found in the database.", nodeId, location, ipAddress);
+    private Optional<NodeDocument> getNodeInfo(final String location, final InetAddress ipAddress, final ContextKey contextKey, final String value) {
+        OnmsNode onmsNode = null;
+
+        if (contextKey != null && !Strings.isNullOrEmpty(value)) {
+            final List<OnmsNode> nodes = nodeDao.findNodeWithMetaData(contextKey.getContext(), contextKey.getKey(), value);
+
+            if (!nodes.isEmpty()) {
+                onmsNode = nodes.get(0);
```

Comment:
If we're using the first node, we should ensure that the results of `findNodeWithMetaData` are consistently ordered. (The order is not deterministic currently.)

Reply:
done